src/cephadm/cephadm.py

```diff
@@ -3512,8 +3512,15 @@ def add_stop_actions(f: TextIO) -> None:
                         '--no-systemd',
                     ]
                 else:
+                    osd_type = 'bluestore'
+                    config_file = '/etc/ceph/ceph.conf'
```

Comment:
Using the `/etc/ceph/ceph.conf` on the host for deciding the objectstore to use is sort of going away from how cpehadm typically handles these things. I think it would be better if this was added as an option to the `objectstore` parameter for OSD specs (https://docs.ceph.com/en/quincy/cephadm/services/osd/#ceph.deployment.drive_group.DriveGroupSpec.objectstore) and then passed as some kind of argument from when the cephadm mgr module makes the deploy call (in `_create_daemon` in serve.py https://github.com/ceph/ceph/blob/main/src/pybind/mgr/cephadm/serve.py#L1253). It would require having `_create_daemon` function check for it when it's handling an OSD and probably require adding a `--objectstore` flag to the cephadm binary `deploy` command as well.

Reply:
Thanks for the comment, the thought is pretty good. But crimson needs to read local_conf, it means that I could change to get objectstore from spec instead of conf file, but I still need to use 'set-extra-ceph-conf' to change conf file, is it OK?

Reply:
Hi, I have modified, please review.

Reply:
As for the conf crimson reads, I expect it will be the one inside the OSD container and not the one actually at /etc/ceph/ceph.conf on the host. For deploying OSDs priort to your patch we already include a conf file at /var/lib/ceph/<fsid>/osd.<osd-id>/config and then mount it at /etc/ceph/ceph.conf in the container, so I think these new OSDs should be able to work the same way.

Reply:
the ceph conf the osd mounts is generated by cephadm so in theory we should be able to modify it if need be as part of the osd deployment. I think this would require implementing a `generate_config` method for the OSDService class (pybind/mgr/cephadm/osd.py). That function would probably have to call the parent class version (`super().generate_config()`) and then add the objectstore setting to to the ceph conf that is generated (possibly making use of the work from https://github.com/ceph/ceph/pull/50318/commits/0d1596e9275e8ca1b8c09b1490b3e595130aa97b)

Reply:
Thanks for your help. I choose to do these all in serve.py, because in my sample yaml file, 'objectstore' is under spec section which is DriveGroupSpec, it's easier to handle all operation in a place. For now, no need to use `set-extra-ceph-conf`, only apply yaml.